data/items.lua

```diff
@@ -799,6 +800,18 @@ function items:GetCategory(data)
   return category
 end
 
+---@param itemLink string
+---@return Enum.ItemBind?
+function items:GetBindTypeFromLink(itemLink)
+  -- itemLink has better information for items, but no information for pet or keystone links
+  local bindType = nil
+  if ( strfind(itemLink, "item:") ) then
+    bindType, _, _, _ = select(14,C_Item.GetItemInfo(itemLink ))
```

Comment:
this needs a bigger refactor to genericly handle non-'item:' links with a fallback to itemID for pets and keys except for any of itemInfo's properties that aren't going to be replace by C_Item.foo APIs

I see this eventually becoming a ---@return itemInfo function

Reply:
you can just do
```
local bindType, _, _, _ = select(14,C_Item.GetItemInfo(itemLink ))
if bindType == nil then
    bindType, _, _, _ = select(14,C_Item.GetItemInfo(itemID ))
end
```

Reply:
> you can just do
I can actually just do
bindType, _, _, _ = select(14,C_Item.GetItemInfo(itemLink ))

as the main AttachItemInfo now handles a falsy result,
but as I noted, this is a placeholder for a bigger rework